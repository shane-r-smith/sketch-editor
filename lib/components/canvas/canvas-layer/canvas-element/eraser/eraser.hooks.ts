import type { LineConfig } from "konva/lib/shapes/Line";
import { useMemo } from "react";
import type { CanvasElementProps } from "../canvas-element.types";

export const createEraser = ({
  element,
  ...rest
}: CanvasElementProps): LineConfig => ({
  ...rest,
  ...element,
  tension: 0.5,
  lineCap: "round",
  lineStroke: "round",
  globalCompositeOperation: "destination-out",
});

export const useEraser = (props: CanvasElementProps): LineConfig => {
  return useMemo<LineConfig>(() => {
    return createEraser(props);
  }, [props]);
};
