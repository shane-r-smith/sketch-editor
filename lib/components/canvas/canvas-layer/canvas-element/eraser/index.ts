export * from "./eraser";
export * from "./eraser.hooks";
