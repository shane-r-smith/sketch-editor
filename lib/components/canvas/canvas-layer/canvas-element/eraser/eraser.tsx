import { Line } from "react-konva";
import { useEraser } from "./eraser.hooks";
import type { CanvasElementProps } from "../canvas-element.types";

export function Eraser(props: CanvasElementProps) {
  const eraser = useEraser(props);

  return <Line {...eraser} />;
}
