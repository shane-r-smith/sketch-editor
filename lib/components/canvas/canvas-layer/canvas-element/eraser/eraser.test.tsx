import { render } from "@testing-library/react";
import { Eraser } from "./eraser";
import type { CanvasElementProps } from "../canvas-element.types";
import { vi } from "vitest";
import type { LineConfig } from "konva/lib/shapes/Line";

vi.mock("react-konva", () => ({
  Line: (props: LineConfig) => (
    <div id="mocked-line">{JSON.stringify(props, undefined, 2)}</div>
  ),
}));

describe("<Eraser />", () => {
  const defaultProps: CanvasElementProps = {
    element: {
      key: "PEN",
      points: [0, 0, 100, 100],
      stroke: "#000000",
      strokeWidth: 10,
      strokeColourType: "HEX",
    },
  };

  it("renders correctly and matches snapshot", () => {
    const { container } = render(<Eraser {...defaultProps} />);
    expect(container).toMatchSnapshot();
  });
});
