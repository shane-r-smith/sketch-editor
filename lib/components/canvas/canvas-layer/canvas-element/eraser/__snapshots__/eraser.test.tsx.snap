// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Eraser /> > renders correctly and matches snapshot 1`] = `
<div>
  <div
    id="mocked-line"
  >
    {
  "tool": "ERASER",
  "points": [
    0,
    0,
    100,
    100
  ],
  "stroke": "#000000",
  "strokeWidth": 10,
  "tension": 0.5,
  "lineCap": "round",
  "lineStroke": "round",
  "globalCompositeOperation": "destination-out"
}
  </div>
</div>
`;
